app/components/TableDateSelector.tsx

```diff
@@ -140,6 +142,28 @@ const getDateRange = (
   return dateRange;
 };
 
+/**
+ * Create a new queryConfig.dateFilter based on the provided filter mode
+ */
+export const makeDateFilterFromMode = (
+  mode: DateFilterMode,
+  queryConfig: QueryConfig,
+  dateColumnName: string
+) => {
+  const dateRange = getDateRange(mode, queryConfig);
+
+  const newFilters = makeDateFilters(dateColumnName || "", dateRange);
+
+  return {
+    column: dateColumnName,
+    mode,
+    filters: newFilters,
+  };
+};
```

Comment:
i isolated this bit of logic into a function that we can pull into the TableWrapper use effect